tests/pipelines/test_pipelines_visual_question_answering.py

```diff
@@ -0,0 +1,147 @@
+        question = "How many cats are there?"
+
+        with self.assertRaises(ValueError):
+            qa(image=[image1, image2], question=[question, question, question])
+            qa(image=[image1], question=[question, question])
+            qa(image=[image1], question=[question, question])
+            qa({"random_key1": image1, "random_key2": question})
+
+
+@is_pipeline_test
+@require_vision
+class VisualQuestionAnsweringPipelineTests(unittest.TestCase, metaclass=PipelineTestCaseMeta):
+    model_mapping = MODEL_FOR_VISUAL_QUESTION_ANSWERING_MAPPING
+
+    def get_test_pipeline(self, model, tokenizer, feature_extractor):
+        vqa_pipeline = pipeline("visual-question-answering", model="sijunhe/tiny-vilt-random-vqa")
+        examples = [
+            {
+                "image": Image.open("./tests/fixtures/tests_samples/COCO/000000039769.png"),
+                "question": "How many cats are there?",
+            },
+            {
+                "image": "http://images.cocodataset.org/val2017/000000039769.jpg",
+                "question": "How many cats are there?",
+            },
+        ]
+        return vqa_pipeline, examples
+
+    def run_pipeline_test(self, vqa_pipeline, examples):
+        outputs = vqa_pipeline(examples, top_k=1)
+        self.assertEqual(
+            outputs,
+            [
+                [{"score": ANY(float), "label": ANY(str)}],
+                [{"score": ANY(float), "label": ANY(str)}],
+            ],
+        )
+
+    @require_torch
+    def test_small_model_pt(self):
+        vqa_pipeline = pipeline("visual-question-answering", model="sijunhe/tiny-vilt-random-vqa")
+        image = "./tests/fixtures/tests_samples/COCO/000000039769.png"
+        query = "How many cats are there?"
+
+        outputs = vqa_pipeline(image=image, question=query, top_k=2)
+        self.assertEqual(
+            outputs, [[{"score": ANY(float), "label": ANY(str)}, {"score": ANY(float), "label": ANY(str)}]]
+        )
+
+        outputs = vqa_pipeline(image=[image, image], question=[query, query], top_k=2)
+        self.assertEqual(
+            outputs, [[{"score": ANY(float), "label": ANY(str)}, {"score": ANY(float), "label": ANY(str)}]] * 2
+        )
+
+    @slow
+    @require_torch
+    def test_large_model_pt(self):
+        vqa_pipeline = pipeline("visual-question-answering", model="dandelin/vilt-b32-finetuned-vqa")
+        image = "./tests/fixtures/tests_samples/COCO/000000039769.png"
+        query = "How many cats are there?"
+
+        outputs = vqa_pipeline(image=image, question=query, top_k=2)
+        self.assertEqual(
+            outputs, [[{"score": ANY(float), "label": ANY(str)}, {"score": ANY(float), "label": ANY(str)}]]
+        )
+
+        outputs = vqa_pipeline(image=[image, image], question=[query, query], top_k=2)
+        self.assertEqual(
+            outputs, [[{"score": ANY(float), "label": ANY(str)}, {"score": ANY(float), "label": ANY(str)}]] * 2
+        )
```

Comment:
Is there a reason we aren't checking the actual predicted answer here?

Reply:
I added the real expected outputs for model `"dandelin/vilt-b32-finetuned-vqa"` and kept the type check for the dummy model `sijunhe/tiny-vilt-random-vqa`